src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Unix.cs

```diff
@@ -20,21 +23,33 @@ public sealed partial class RuntimeThread
 
         internal WaitSubsystem.ThreadWaitInfo WaitInfo => _waitInfo;
 
+#if MONO
+        private void Unix_PlatformSpecificInitialize()
```

Comment:
The idea is you would have a `win32_net_4_x_corlib.dll.sources` for the net_4_x profile on win32 for example, or `darwin_testing_aot_hybrid.corlib.dll.sources` for testing_aot_hybrid profile on darwin (== macOS).